packages/hardhat-core/src/internal/artifacts.ts

```diff
@@ -469,18 +469,21 @@ export function getArtifactFromContractOutput(
   contractName: string,
   contractOutput: any
 ): Artifact {
-  const evmBytecode = contractOutput.evm && contractOutput.evm.bytecode;
+  const evmBytecode =
+    Boolean(contractOutput.evm) && contractOutput.evm.bytecode;
   let bytecode: string =
-    evmBytecode && evmBytecode.object ? evmBytecode.object : "";
+    Boolean(evmBytecode) && Boolean(evmBytecode.object)
```

Comment:
Same comment here and in the rest of the changes that use `Boolean` (we should use `?.` and `??` instead).

Reply:
I may be wrong but line 472 seems like a bug since it should be returning an object but it is returning a boolean.